internal/premis/premis.go

```diff
@@ -0,0 +1,447 @@
+package premis
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+	"time"
+
+	"github.com/beevik/etree"
+	"github.com/google/uuid"
+
+	"github.com/artefactual-sdps/preprocessing-sfa/internal/fsutil"
+)
+
+const EmptyPremis = `<?xml version="1.0" encoding="UTF-8"?>
+<premis:premis xmlns:premis="http://www.loc.gov/premis/v3" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.loc.gov/premis/v3 https://www.loc.gov/standards/premis/premis.xsd" version="3.0">
+</premis:premis>
+
+`
+
+type PREMISObjectEventIdentifier struct {
+	IdType  string
+	IdValue string
+}
+
+type PREMISObject struct {
+	IdType           string
+	IdValue          string
+	OriginalName     string
+	EventIdentifiers []PREMISObjectEventIdentifier
+}
+
+type PREMISEventSummary struct {
+	Type    string
+	Detail  string
+	Outcome string
+}
+
+type PREMISEvent struct {
+	IdType       string
+	IdValue      string
+	Summary      PREMISEventSummary
+	DateTime     string
+	AgentIdType  string
+	AgentIdValue string
+}
+
+type PREMISAgent struct {
+	IdType  string
+	IdValue string
+	Name    string
+	Type    string
+}
+
+func PREMISAgentDefault() PREMISAgent {
+	return PREMISAgent{
+		Type:    "software",
+		Name:    "Enduro",
+		IdType:  "url",
+		IdValue: "https://github.com/artefactual-sdps/preprocessing-base",
+	}
+}
+
+func ParseFile(filePath string) (*etree.Document, error) {
+	doc := etree.NewDocument()
+
+	err := doc.ReadFromFile(filePath)
+	if err != nil {
+		return nil, err
```

Comment:
It would be helpful to add a bit of context to the error message here (and in general) to help a developer encountering the error to find where it originated.

e.g.
```go
return nil, fmt.Errorf("parse XML: %v")
```